src/database.rs

```diff
@@ -59,6 +59,15 @@ pub trait Database: Send + ZalsaDatabase + AsDynDatabase {
         zalsa_mut.runtime_mut().report_tracked_write(durability);
     }
 
+    /// This method triggers cancellation.
+    /// If you invoke it while a snapshot exists, it
+    /// will block until that snapshot is dropped -- if that snapshot
+    /// is owned by the current thread, this could trigger deadlock.
+    #[cfg(feature = "salsa_unstable")]
```

Comment:
I think it's fine to commit this as stable API. It seems generally useful to me to be able to trigger cancellation without bumping the revision or triggering LRU garbage collection.

Also not like salsa's API is generally stable right now (as is evident by me having removed that function 🙈)